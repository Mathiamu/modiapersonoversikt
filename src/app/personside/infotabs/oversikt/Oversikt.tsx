import * as React from 'react';
import styled from 'styled-components';
import Oversiktskomponent from './Oversiktskomponent';
import { pxToRem } from '../../../../styles/personOversiktTheme';
import VarselOversikt from './VarselOversikt';
import OppfolgingOversikt from './OppfolgingOversikt';
import YtelserOversikt from './YtelserOversikt';
import UtbetalingerOversikt from './UtbetalingerOversikt';
import MeldingerOversikt from './MeldingerOversikt';

const Style = styled.article`
    display: flex;
`;

const KolonneStyle = styled.div`
    > * {
        margin: ${pxToRem(15)};
        max-width: 30rem;
    }
`;

function Oversikt() {
    return (
        <Style>
            <KolonneStyle>
                <Oversiktskomponent
                    uriKomponent={'utbetaling'}
                    tittel={'Utbetalinger'}
                    children={<UtbetalingerOversikt />}
                />
                <Oversiktskomponent
                    uriKomponent={'oppfølging'}
                    tittel={'Oppfølging'}
                    children={<OppfolgingOversikt />}
                />
                <Oversiktskomponent uriKomponent={'varsler'} tittel={'Varsler'} children={<VarselOversikt />} />
            </KolonneStyle>
            <KolonneStyle>
                <Oversiktskomponent uriKomponent={'meldinger'} tittel={'Meldinger'} children={<MeldingerOversikt />} />
                <Oversiktskomponent uriKomponent={'ytelser'} tittel={'Ytelser'} children={<YtelserOversikt />} />
            </KolonneStyle>
        </Style>
    );
}

export default Oversikt;
