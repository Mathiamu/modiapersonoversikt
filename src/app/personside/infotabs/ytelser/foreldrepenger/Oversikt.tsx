import * as React from 'react';
import { Foreldrepengerrettighet } from '../../../../../models/ytelse/foreldrepenger';
import theme from '../../../../../styles/personOversiktTheme';
import { utledMaksDato, utledFraDatoForRettighet } from './foreldrePengerUtils';
import { datoVerbose } from '../../utbetalinger/utils/utbetalingerUtils';
import styled from 'styled-components';
import DescriptionList from '../DescriptionList';
import YtelserBullet from '../YtelserBullet';

interface Props {
    foreldrePenger: Foreldrepengerrettighet;
}

const Wrapper = styled.div`
    margin: 0 ${theme.margin.px20} ${theme.margin.px20};
  > *:not(:first-child) {
    border-top: ${theme.border.skilleDashed};
  }
`;

function Oversikt({foreldrePenger}: Props) {

    const foreldrePengeRetten = {
        Foreldrepengetype: foreldrePenger.foreldrepengetype,
        Dekningsgrad: foreldrePenger.dekningsgrad + '%',
        Dagsats: 'Ikke implementert',
        Graderingsdager: 'Ikke implementert',
        'Fedrekvote tilogmed': 'Ikke implementert',
        'Mødrekvote tilogmed': 'Ikke implementert',
        'Rettighet fra dato': datoVerbose(utledFraDatoForRettighet(foreldrePenger)).sammensatt,
        Restdager: foreldrePenger.restDager,
        Maksdato: utledMaksDato(foreldrePenger),
        Arbeidskategori: 'Ikke implementert'
    };

    const barnet = {
        Termindato: foreldrePenger.rettighetFom && datoVerbose(foreldrePenger.rettighetFom).sammensatt,
        Fødselsdato: foreldrePenger.barnetsFødselsdato && datoVerbose(foreldrePenger.barnetsFødselsdato).sammensatt,
        'Annen forelder': foreldrePenger.andreForeldersFnr,
        Omsorgsovertakelse: 'Ikke implementert',
        'Foreldre av samme kjønn': 'Ikke implementert',
        'Antall barn': foreldrePenger.antallBarn
    };

    return (
        <Wrapper>
            <YtelserBullet tittel="Om foreldrepengeretten">
                <DescriptionList entries={foreldrePengeRetten}/>
            </YtelserBullet>
            <YtelserBullet tittel="Om barnet">
                <DescriptionList entries={barnet}/>
            </YtelserBullet>
        </Wrapper>
    );
}

export default Oversikt;
