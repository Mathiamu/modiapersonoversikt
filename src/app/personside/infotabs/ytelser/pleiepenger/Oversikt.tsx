import * as React from 'react';
import { Pleiepengerettighet } from '../../../../../models/ytelse/pleiepenger';
import { OversiktStyling } from '../felles-styling/CommonStylingYtelser';
import YtelserBullet from '../felles-styling/YtelserBullet';
import DescriptionList from '../felles-styling/DescriptionList';
import ForbrukteDager from './ForbrukteDager';

interface Props {
    pleiepenger: Pleiepengerettighet;
}

function Oversikt({pleiepenger}: Props) {

    const omPleiepengerettenEntries = {
        'Fra og med': 'Ikke implementert',
        'Til og med': 'Ikke implementert',
        Kompensasjonsgrad: 'Ikke implementert',
        Pleiepengegrad: 'Ikke implementert',
        'Totalt invilget': 'Ikke implementert',
        Ferieperioder: 'Ikke implementert',
        Barnet: pleiepenger.barnet,
        'Annen forelder': pleiepenger.andreOmsorgsperson
    };

    const arbeidsSitsuasjonEntries = {
        Arbeidsgiver: '',
        Arbeidskategori: '',
        Inntekstsperiode: '',
        Kontonummer: '',
        Refusjonstype: '',
        'Inntekt for perioden': '',
        'Refusjon til dato': ''
    };

    return (
        <OversiktStyling>
            <YtelserBullet tittel="Om pleiepengeretten">
                <ForbrukteDager pleiepenger={pleiepenger}/>
                <DescriptionList entries={omPleiepengerettenEntries}/>
            </YtelserBullet>
            <YtelserBullet tittel="Arbeidssituasjon">
                <DescriptionList entries={arbeidsSitsuasjonEntries}/>
            </YtelserBullet>
        </OversiktStyling>);
}

export default Oversikt;
