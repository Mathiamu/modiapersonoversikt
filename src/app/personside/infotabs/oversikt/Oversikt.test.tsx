import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../test/Testprovider';
import Oversikt from './Oversikt';

test('Viser oversikt med alt innhold', () => {
    const container = renderer.create(
        <TestProvider>
            <Oversikt />
        </TestProvider>
    );

    const json = container.toJSON();
    expect(json).toMatchSnapshot();
});
