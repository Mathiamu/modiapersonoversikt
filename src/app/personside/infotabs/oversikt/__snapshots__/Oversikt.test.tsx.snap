// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser oversikt med alt innhold 1`] = `
.c7 {
  font-weight: bold;
}

.c2 {
  border-radius: 0.25rem;
  background-color: white;
  box-shadow: 0 0 0.5rem rgba(0,0,0,0.1);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.625rem;
}

.c4 > * {
  border-top: solid 0.0625rem #b7b1a9;
}

.c9 > *:not(:first-child) {
  border-top: solid 0.0625rem #b7b1a9;
}

.c10 {
  display: grid;
  grid-template-columns: 70% auto;
  background-color: white;
  padding: 0.625rem;
}

.c11 li {
  display: inline-block;
}

.c11 li:not(:last-child):after {
  content: ',';
  margin-right: 0.5em;
}

.c8 {
  padding: 0.625rem;
}

.c12 > *:not(:first-child) {
  border-top: solid 0.0625rem #b7b1a9;
}

.c12 > * {
  padding: 0.625rem;
}

.c5 > *:not(:first-child) {
  border-top: solid 0.0625rem #b7b1a9;
}

.c6 {
  background-color: white;
  padding: 0.625rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 > * {
  margin: 0.9375rem;
  max-width: 30rem;
}

<article
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <h3
          className="typo-undertittel"
        >
          Utbetalinger
        </h3>
        <a
          className="lenke"
          href="/modiapersonoversikt/person/10108000398/utbetaling/"
          onClick={[Function]}
        >
          <p
            className="typo-normal"
          >
            Gå til 
            utbetalinger
          </p>
        </a>
      </div>
      <div
        className="c4"
      >
        <ol
          className="c5"
        >
          <div
            className="c6"
          >
            <p
              className="typo-normal"
            >
              26.03.2017
               / 
              Utbetalt
            </p>
            <p
              className="typo-normal"
            >
              <span
                className="c7"
              >
                Sykepenger
              </span>
            </p>
            <p
              className="typo-normal"
            >
              09.11.2016
               - 
              31.10.2017
            </p>
            <p
              className="typo-normal"
            >
              Utbetaling til: 
              AREMARK TESTFAMILIEN
            </p>
          </div>
        </ol>
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <h3
          className="typo-undertittel"
        >
          Oppfølging
        </h3>
        <a
          className="lenke"
          href="/modiapersonoversikt/person/10108000398/oppfølging/"
          onClick={[Function]}
        >
          <p
            className="typo-normal"
          >
            Gå til 
            oppfølging
          </p>
        </a>
      </div>
      <div
        className="c4"
      >
        <div
          className="c8"
        >
          <p
            className="typo-normal"
          >
            Er ikke i arbeidsrettet oppfølging
          </p>
        </div>
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <h3
          className="typo-undertittel"
        >
          Varsler
        </h3>
        <a
          className="lenke"
          href="/modiapersonoversikt/person/10108000398/varsler/"
          onClick={[Function]}
        >
          <p
            className="typo-normal"
          >
            Gå til 
            varsler
          </p>
        </a>
      </div>
      <div
        className="c4"
      >
        <ol
          className="c9"
        >
          <div
            className="c10"
          >
            <p
              className="typo-normal"
            >
              28. mars 2019
            </p>
            <ul
              aria-label="Kommunikasjonskanaler"
              className="c11"
            >
              <li
                className="typo-normal"
              >
                SMS
              </li>
              <li
                className="typo-normal"
              >
                EPOST
              </li>
            </ul>
            <p
              className="typo-normal"
            >
              <span
                className="c7"
              >
                Påminnelse om å sende meldekort
              </span>
            </p>
          </div>
          <div
            className="c10"
          >
            <p
              className="typo-normal"
            >
              10. jan. 2019
            </p>
            <ul
              aria-label="Kommunikasjonskanaler"
              className="c11"
            >
              <li
                className="typo-normal"
              >
                EPOST
              </li>
            </ul>
            <p
              className="typo-normal"
            >
              <span
                className="c7"
              >
                Påminnelse om å sende meldekort
              </span>
            </p>
          </div>
        </ol>
      </div>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <h3
          className="typo-undertittel"
        >
          Meldinger
        </h3>
        <a
          className="lenke"
          href="/modiapersonoversikt/person/10108000398/meldinger/"
          onClick={[Function]}
        >
          <p
            className="typo-normal"
          >
            Gå til 
            meldinger
          </p>
        </a>
      </div>
      <div
        className="c4"
      />
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <h3
          className="typo-undertittel"
        >
          Ytelser
        </h3>
        <a
          className="lenke"
          href="/modiapersonoversikt/person/10108000398/ytelser/"
          onClick={[Function]}
        >
          <p
            className="typo-normal"
          >
            Gå til 
            ytelser
          </p>
        </a>
      </div>
      <div
        className="c4"
      >
        <div
          className="c12"
        />
      </div>
    </div>
  </div>
</article>
`;
